flocker/control/test/test_protocol.py

```diff
@@ -679,26 +729,60 @@ def test_configuration_change(self):
         """
         agent = FakeAgent()
         client = AgentAMP(Clock(), agent)
-        service = build_control_amp_service(self)
+        service_clock = Clock()
+        service = build_control_amp_service(self, service_clock)
         service.startService()
         server = LoopbackAMPClient(client.locator)
         service.connected(server)
 
         service.configuration_service.save(TEST_DEPLOYMENT)
+        service_clock.advance(10.0)
 
         self.assertEqual(
             dict(configuration=TEST_DEPLOYMENT, state=DeploymentState()),
             dict(configuration=agent.desired, state=agent.actual),
         )
 
+    def test_coalesced_sends_within_time(self):
+        """
+        Sending updates multiple times within a second only actually causes 1
+        update to be sent to the agent.
+        """
+        agent = FakeAgent()
+        client = AgentAMP(Clock(), agent)
+        service_clock = Clock()
+        service = build_control_amp_service(self, service_clock)
+        service.startService()
+        server = LoopbackAMPClient(client.locator)
+        service.connected(server)
+
+        initial_updates_count = agent.cluster_updated_count
+        for i in xrange(10):
+            service.configuration_service.save(
+                TEST_DEPLOYMENT.transform(
+                    ['nodes'],
+                    lambda nodes: nodes.union(frozenset([
+                        Node(hostname='%d.example.com' % i)
+                    ]))
```

Comment:
That's much better. But I also just stumbled upon `arbitrart_tranformation` which conveys what I want.